ufl/algorithms/ad.py

```diff
@@ -27,6 +28,12 @@ def expand_derivatives(form, **kwargs):
     if kwargs:
         warning("Deprecation: expand_derivatives no longer takes any keyword arguments")
 
+    if isinstance(form, Adjoint):
+        dform = expand_derivatives(form._form)
+        if dform == 0:
+            return dform
+        raise NotImplementedError('Adjoint derivative is not supported.')
```

Comment:
Put a comment in here reminding us that this is a 3-form can't happen situation.

Reply:
Done